src/functions-and-arrays.js

```diff
@@ -1,19 +1,119 @@
 // Iteration #1: Find the maximum
+function maxOfTwoNumbers (a, b) {
+  if (a > b) {
+    return a;
+  }
+  return b
+};
+
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
+
+
+function findLongestWord(arr) {
+  
+
+  if (typeof arr == 'undefined' || arr.length < 1){
+    return null;
+  } 
+
+  let longestWord = 'a';
+
+  for (let i = 0; i < arr.length; i++) {
+    if (arr[i].length > longestWord.length) {
+      longestWord = arr[i];
+    }
+  }
+  return longestWord;
+}
+
+
+
+
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+
+
+function sumNumbers(arr) {
+
+  if (typeof arr == 'undefined' || arr.length < 1){
+    return 0;
+  } 
+
+  let sum = 0;
+
+  for (let i = 0; i < arr.length; i++) {
+    sum += arr[i];
+  }
+  return sum;
+}
+
+
+//Bonus - Iteration #3.1: A generic sum() function
+
+function sum(arr) {
+
+  if (typeof arr == 'undefined' || arr.length < 1){
+    return 0;
+  } 
+
+  let sum = 0;
+
+  for (let i = 0; i < arr.length; i++) {
+    
+    if (typeof arr[i] === 'number') {
```

Comment:
you can actually have an or satetment here: 
typeof arr[i] === 'number' || arr[i] == true

because: 5 + true === 6